subsys/net/ip/net_stats.h

```diff
@@ -387,6 +409,34 @@ static inline void net_stats_update_tc_tx_time(struct net_if *iface,
 }
 #endif /* CONFIG_NET_CONTEXT_TIMESTAMP && CONFIG_NET_STATISTICS */
 
+#if defined(CONFIG_NET_PKT_RXTIME_STATS) && defined(CONFIG_NET_STATISTICS) \
+	&& defined(CONFIG_NET_NATIVE)
+static inline void net_stats_update_tc_rx_time(struct net_if *iface,
+					       u8_t tc,
+					       u32_t start_time,
+					       u32_t end_time)
+{
+	u32_t diff = abs(end_time - start_time);
+
+	UPDATE_STAT(iface, stats.tc.recv[tc].rx_time.time_sum +=
+		    SYS_CLOCK_HW_CYCLES_TO_NS64(diff) / 1000);
+	UPDATE_STAT(iface, stats.tc.recv[tc].rx_time.time_count += 1);
+
+	net_stats_update_rx_time(iface, start_time, end_time);
+}
+#else
+static inline void net_stats_update_tc_rx_time(struct net_if *iface,
```

Comment:
now that I look at the signature, maybe it would be much simpler by providing only one parameter: net_pkt.
From here you can get the iface, its priority, the start timestamp, the end can be directly got from k_cycle_get(). 

and that will help for (see below):

Reply:
The RX API is done similar way as the TX one and I would like to keep them like that. For the TX one, the net_pkt is freed before the stats function is called, and I would like not to have net_pkt_ref() called as it keeps net_pkt in use for longer and is generally pain to debug.
But I think we can overcome this by not using inline function for the update function, if we have just a define, then the parameters will just disappear and we can avoid the ifdef's. I will investigate this route more.

Reply:
indeed, I checked tx side and that would need a bit of rework to get to a simpler signature. As you want then.
That said, why are we counting tx stats only if there is a context attached to it? (does that mean we calculate the stats only on user packets?)

Reply:
The context time counting was done earlier, the idea was that application could activate statistics counting for just one socket. With this new config option, we can get the statistics for all the sockets. Initially I thought that we could remove the context time counting but after some more thinking, it might make sense to allow the application to gather TX statistics for just one socket so we could keep this functionality. The code looks a bit ugly in places (mainly in net_pkt.h) because of various ifdef's, but we can probably live with that.